### PR TITLE
File handling program 

### PR DESCRIPTION
**Pull request by 31673**
Created new file handling program

### CODE REVIEW

File: linkedlist.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+newNode=newNode->next;
+}
+printf("\nTotal nodes:%d\n",count);
+}
+
+int main()
+{
+int choice;
+ void (*ptr [3])()={ insertion_beginning, display,exit};
+while(1)
```
Comment (P-Prajwal) [inline linkedlist.c]: What about "exit" option? Shouldn't the loop break on this case?

**Review by P-Prajwal — CHANGES_REQUESTED** [review]

File: filehandling.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#include<stdio.h>
+
+struct employee
+{
```
Comment (preeja398) [inline filehandling.c:4]: name of struct, function etcc whould start with Capital letter

struct Employee

Brackets format should be as

struct Employee {
}

If {
}

File: filehandling.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	printf("error in open file in write mode");
+}
+else
+{
+	for(i=0;i<2;i++)
+ fwrite(&emp[i],sizeof(struct employee),1,fp);
+}
+
+fp=fopen("file.bin","rb");
+{
```
Comment (tklogesh) [inline filehandling.c:35]: '{' should be inside if statement

File: filehandling.c [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+int main()
+{
+	FILE *fp=NULL;
+	struct employee emp[2];
+	int i;
+	for(i=0;i<2;i++)
+{
+	printf("Enter the employee details\n");
+	printf("Name:");
+	scanf("%s",emp[i].employeeName);
```
Comment (preeja398) [inline filehandling.c:18]: what will happen when I enter name whose length > 50 ?

File: filehandling.c [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+struct employee
+{
+	char employeeName[50];
+	int employeeId;
+};
+
+int main()
+{
+	FILE *fp=NULL;
```
Comment (preeja398) [inline filehandling.c:11]: add space before and after each operator File* fp = NULL;

File: filehandling.c [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	struct employee emp[2];
+	int i;
```
Comment (preeja398) [inline filehandling.c:13]: give some valid name like int index;

**Comment by tklogesh** [comment]
Align the code in proper format.

File: filehandling.c [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+	for(i=0;i<2;i++)
```
Comment (preeja398) [inline filehandling.c:14]: add space before operators like

for (i = 0; i < 2; i++) {
}

File: filehandling.c (hunk 4 above)
Comment (preeja398) [inline filehandling.c:18]: Add space after ,
scanf("%s", emp[i].employeeName);

File: filehandling.c [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+{
+	printf("Enter the employee details\n");
+	printf("Name:");
+	scanf("%s",emp[i].employeeName);
+	printf("Id:");
+	scanf("%d",&emp[i].employeeId);
+}
+
+fp=fopen("file.bin","wb");
```
Comment (preeja398) [inline filehandling.c:23]: space before and after operators

File: filehandling.c (hunk 3 above)
Comment (HKilaji) [reply filehandling.c:35]: Indentation Issue:  try to use same format for each lines, either use Tab or Space. Don't use both in same code.
Also make code follow same format for each line inside a scope.
some thing like this:
```
if (fp == NULL) {  
    printf("error in open file in write mode");
} else {
    for (i = 0; i < 2; i++) {
         fwrite(&emp[i], sizeof(struct employee), 1, fp);
    }
}

fp = fopen("file.bin", "rb");
{
```

File: filehandling.c [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+else
+{
+	for(i=0;i<2;i++)
+ fwrite(&emp[i],sizeof(struct employee),1,fp);
+}
+
+fp=fopen("file.bin","rb");
+{
+	if(fp==NULL)
+	printf("error open filn in read mode");
```
Comment (HKilaji) [inline filehandling.c:37]: Same as above.

File: filehandling.c [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+{
+for(i=0;i<2;i++)
+{
+fread(&emp[i],sizeof(struct employee),1,fp);
+printf("Employee name:%s\tEmployee id:%d\n",emp[i].employeeName,emp[i].employeeId);
+}
+}
+}
+}
+
```
Comment (HKilaji) [inline filehandling.c:48]: Opening and closing of scopes should follow same format. 
as shown above

File: filehandling.c [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+{
+	printf("error in open file in write mode");
+}
+else
+{
+	for(i=0;i<2;i++)
+ fwrite(&emp[i],sizeof(struct employee),1,fp);
+}
+
+fp=fopen("file.bin","rb");
```
Comment (P-Prajwal) [inline filehandling.c:34]: Always close the file before opening it again.

```suggestion
fclose(fp);

fp=fopen("file.bin","rb");
```

File: filehandling.c [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+	if(fp==NULL)
+	printf("error open filn in read mode");
+else
+{
+for(i=0;i<2;i++)
+{
+fread(&emp[i],sizeof(struct employee),1,fp);
+printf("Employee name:%s\tEmployee id:%d\n",emp[i].employeeName,emp[i].employeeId);
+}
+}
```
Comment (P-Prajwal) [inline filehandling.c:45]: Close file before exiting else.

File: filehandling.c (hunk 3 above)
Comment (P-Prajwal) [inline filehandling.c:35]: Why this curly brace is opened here? Please remove this.

File: filehandling.c (hunk 2 above)
Comment (31673) [reply filehandling.c:4]: ok

File: filehandling.c (hunk 3 above)
Comment (31673) [reply filehandling.c:35]: I changed as your review commands . can you please verify once

File: filehandling.c (hunk 4 above)
Comment (31673) [reply filehandling.c:18]: It's illegal or error sometimes it will crash the program

File: filehandling.c (hunk 5 above)
Comment (31673) [reply filehandling.c:11]: I done changes as your review commands . can you please verify once

File: filehandling.c (hunk 6 above)
Comment (31673) [reply filehandling.c:13]: ok

File: filehandling.c (hunk 10 above)
Comment (31673) [reply filehandling.c:48]: ok

File: filehandling.c (hunk 11 above)
Comment (31673) [reply filehandling.c:34]: I done changes as your review commands . can you please verify once

File: filehandling.c (hunk 12 above)
Comment (31673) [reply filehandling.c:45]: ok sure

**Pull request closed without merging**